src/app/shared/services/base.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { environment } from '../../../environments/environment';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { catchError, retry } from 'rxjs/operators';
+import { map } from 'rxjs/operators';
+import { Deserializer } from 'ts-jsonapi';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BaseService {
```

Comment:
@carryall How about to make it `abstract` class as this class will be used for inheritance and then no one cant cannot directly create an instance of an abstract class?

Reply:
✅  Fixed on [e8d6d6e](https://github.com/carryall/survey-challenge/pull/31/commits/e8d6d6eec9da251ee7dead5965815559c3d97062)